pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2126,13 +2141,15 @@ public boolean addNewReloadAllSegmentsJob(String tableNameWithType, String jobId
     jobMetadata.put(CommonConstants.ControllerJob.JOB_TYPE, ControllerJobType.RELOAD_ALL_SEGMENTS.toString());
     jobMetadata.put(CommonConstants.ControllerJob.SUBMISSION_TIME_MS, Long.toString(System.currentTimeMillis()));
     jobMetadata.put(CommonConstants.ControllerJob.MESSAGE_COUNT, Integer.toString(numberOfMessagesSent));
-    return addControllerJobToZK(jobId, jobMetadata);
+    return addControllerJobToZK(jobId, jobMetadata,
```

Comment:
When does this znode get cleaned up?

Reply:
We are using PERSISTENT mode for all our Zookeeper updates (not just this one). I dont think this gets up cleaned automatically. Need to check if we have any periodic zookeeper clean up mechanism. 
https://zookeeper.apache.org/doc/r3.1.2/api/org/apache/zookeeper/CreateMode.html#PERSISTENT

Reply:
Looked at the code (addControllerJobToZK). We have an upper bound on how many Znodes we can create under a resource directory (MAXIMUM_CONTROLLER_JOBS_IN_ZK = 100). Beyond this, we remove the older jobs based on submission time.